Tgui Jumbo PR: Fixes, CSS refactoring, new components

stylemistake opened the pull request:
## About The Pull Request

Fixes #47577
Fixes #47649

Massive refactoring of CSS:

- Massively improve performance of CSS compiler (build time went down 18s -> 7s).
- Reduced size of JS (332KB -> 309KB)
- Reduced size of CSS (65KB -> 44KB)
- Nuke all old colors.
- Use a spectrum-based set of colors.
- Tweak colors for best visual appearance in UIs, use a more robust color math, fonts should be always bright and legible.
- Rounded corners in all UI controls.
- Tweak baselines of UI elements.
- Nuke useless vars, rename all vars to more appropriate names.
- Button icons are now fixed width, so buttons no longer change size when toggling them.

Small refactoring of UIs:

- Created a reusable BeakerContents component.
- Improved table layout of Smart Fridge and Crew Console.
- Code style fixes.

New features:

- `ColorBox` component for rendering a small 1-character wide color box.
- `Dimmer` component for dimming whole window pane for showing important content.
- `AnimatedNumber` render prop is now called with two parameters: `(formattedValue, rawValue) => any`.
- `Table.Row` and `Table.Cell` have a `header` property for declaring table headers.
- `ProgressBar` supports the whole range of colors.

Fixes:

- Crafting UI will no longer reset its state when pressing "Craft".
- Computer Fabricator actually works.

Miscellaneous:

- Updated all dependencies to latest versions. This should fix compatibility with Node v13.
- Reintroduced a custom Babel plugin for multiline string compile-time evaluation, which reduces bundle size by about 5-6KB.
- Cherry picked corejs features to reduce bundle size by further 20KB.
- Node v13 compatibility by removing Fibers package without noticeable slowdown.

<br><details>
<summary><b>LOTS AND LOTS OF SCREENSHOTS</b></summary><br>

![image](https://user-images.githubusercontent.com/1516236/68360876-61d7ac80-012a-11ea-8e2f-a20667d0b808.png)

![image](https://user-images.githubusercontent.com/1516236/68361059-0ce86600-012b-11ea-8c27-cf9eae484618.png)

![image](https://user-images.githubusercontent.com/1516236/68361067-170a6480-012b-11ea-9282-d466ff5f9158.png)

![image](https://user-images.githubusercontent.com/1516236/68361104-47ea9980-012b-11ea-9392-6d3972bc8c0f.png)

![image](https://user-images.githubusercontent.com/1516236/68361136-66509500-012b-11ea-82a1-242601477f7d.png)

![image](https://user-images.githubusercontent.com/1516236/68361167-87b18100-012b-11ea-9120-7ae64aece97e.png)

![image](https://user-images.githubusercontent.com/1516236/68366043-f3e7b100-013a-11ea-9633-8e43f7d0f704.png)

![image](https://user-images.githubusercontent.com/1516236/68366628-7329b480-013c-11ea-96c2-41d9df448e07.png)

</details>

## Why It's Good For The Game

It's good, mmkay?

## Changelog
:cl:
fix: Personal Crafting UI will no longer reset its state when crafing an item.
fix: Improved table layout of Smart Fridge and Crew Console.
fix: Fix ORM not smelting alloys.
fix: Fix Computer Fabricator.
refactor: Tgui CSS got a massive rework.
/:cl:

stylemistake committed:
Pass both formatted and raw values in AnimatedNumber custom renderer

stylemistake committed:
Upgrade dependencies, improve build speed

- luminance helper is not so expensive to call now.

stylemistake committed:
Refactor and fix loss of state in PersonalCrafting

stylemistake committed:
Refactor CrewConsole, move dept and damage colors to JS

stylemistake committed:
Rework CrewConsole, add ColorBox

stylemistake committed:
CSS rework

stylemistake committed:
More CSS tweakage

TheVekter commented:
Did you work out drop down menus or is that still WIP?

stylemistake commented:
@TheVekter actioninja is currently working on them.

HomocidlePancakes commented:
Rounded corner are fucking virgin AF. I want my 90° angles back

Ohagi-Chan commented:
Thank goodness. I got the impression crafting wasn't getting fixed.
https://github.com/tgstation/tgstation/issues/47577

stylemistake committed:
Tweak colors

stylemistake commented:
It was easy enough to fix.

stylemistake committed:
Fix Wires UI (fall back to standard html color names)

stylemistake commented on tgui-next/packages/tgui/interfaces/PersonalCrafting.js:
```diff
@@ -88,98 +88,90 @@ export const PersonalCrafting = props => {
+  const craftingRecipes = map((subcategory, category) => {
+    const hasSubcats = ('has_subcats' in subcategory);
+    const firstSubcatName = Object.keys(subcategory)
+      .find(name => name !== 'has_cubcats');
```
it was your typo, and you said it was fine.

actioninja replied:
I don't even remember saying that and you probably shouldn't have listened to me. It should be `has_subcats` in every place it appears.

stylemistake committed:
Parse NaNs and Infinities sent by json_encode proc

stylemistake commented:
Chad Lummox has added support for NaN and Infinity. We can finally observe fusion glitches in full glory! http://www.byond.com/forum/post/2520870

stylemistake commented:
This is more or less final. Please test/review, feel free to merge.

stylemistake committed:
Fix ORM not smelting alloys

stylemistake committed:
Rebuild tgui

stylemistake committed:
Multiple changes

- Fix Grid usage in ComputerFabricator
- Re-add a babel plugin for "multiline", this time compatible with current NodeJS
- Cherry-pick certain corejs features
- Add all currently existing eslint rules with comments
- Add linter check for undefined globals (missing imports)

stylemistake committed:
Bump NodeJS requirement, direct to LTS page

stylemistake committed:
Minor dependency upgrade, remove fibers for better compatibility

SpaceManiac reviewed (changes requested):
This PR breaks Wine support

First problem:
![image](https://user-images.githubusercontent.com/222630/68532891-c2f0b300-02d7-11ea-9343-3c6c8633b95a.png)

Second problem:
![image](https://user-images.githubusercontent.com/222630/68532893-c4ba7680-02d7-11ea-9aec-3bcc86c27679.png)

stylemistake committed:
Fix/screw IE8

stylemistake committed:
Update all IE8-related comments

optimumtact commented:
rounded corners eww

stylemistake committed:
Revert rounding of buttons

stylemistake commented:
To reduce bikeshedding and start solving real issues, I decided to revert some controversial changes.

### Button rounding is gone.

![crab rave](https://i.kym-cdn.com/photos/images/original/001/421/797/f5a.gif)

Now click that juicy **MERGE** button!

stylemistake committed:
Tweak margins around button icons

optimumtact commented:
![](https://camo.githubusercontent.com/69a50acbcc15f9a45ef4afc777fe57184e2187ec/68747470733a2f2f692e6b796d2d63646e2e636f6d2f70686f746f732f696d616765732f6f726967696e616c2f3030312f3432312f3739372f6635612e676966)
